public/static/docs/command-reference/get.md

```diff
@@ -65,6 +65,11 @@ name.
   the file or directory from. The tip of the default branch is used by default
   when this option is not specified.
 
+- `--show-url` - `url` is expected to represent a DVC project for this option to
+  have an effect (as opposed to a Git-only, non-DVC repository). When
+  `--show-url` is used, instead of downloading the file or directory, this
+  command just prints the storage location (URL) of the target data.
```

Comment:
I still think we are overcomplicating this. The fact that it does not handle Git repos is only somewhat important and at least should go after the essence of what this command does. Also, probably it's not about DVC repo vs Git repo - it's about Git-tracked files vs DVC-tracked files. Am I correct here, @skshetry ?

Also, consider including a link the example below.

Reply:
No, it's about DVC repo vs Git repo. As an example, take this *git* repo. How can you get an url for this `public/static/docs/command-reference/get.md`? Git does not work with files, so there's no URL/path to generate for (unless we do somekind of *magic*).

Reply:
okay, to clarify it's not about Git-only vs DVC. Any Git+DVC has files that are not under DVC control and probably we don't generate URL for them, right?

Reply:
I'm not even sure about DVC-controlled non-cached files (`dvc run -O something`), probably we don't even handle that with this new option (even though we can download it)

Reply:
Yes, that should not also work, as it checks for all DVC controlled outputs.

Reply:
So, it's also about Git vs DVC tracked files. Sorry about that.

Reply:
> The fact that it does not handle Git repos is only somewhat important and at least should go after the essence of what this command does.

@shcheklein You may be right but about this we're already using this style in most other options when there are pre-requisites. Putting the pre-req's as the first sentence. (See `--rev` just above this.) That's not a great reason to keep this but I'm also not convinced that leaving the pre-req's for last is best. Maybe it is better to leave them on top? they are "pre" requisites after all 🙂

All that said I did shorten that first sentence now, and also the main explanation following it.

> to clarify it's not about Git-only vs DVC. Any Git+DVC has files that are not under DVC control

True, I updated that as well. So the pre-req is not about `url` then, but about `path`: it should be a DVC-tracked file/dir.

See ab40ed7f.

Reply:
> I'm not even sure about DVC-controlled non-cached files (dvc run -O something), probably we don't even handle that with this new option (even though we can download it)

This sounds like something I should note here and in `dvc.api.get_url` as well?

Reply:
> I'm not even sure about DVC-controlled non-cached files (dvc run -O something), probably we don't even handle that with this new option (even though we can download it)

Actually, both this option and `get_url()` should support creating a URL for these. They don't atually check for the existence of the file in remote cache. They just interpret the project configured remote + the checksum in DVC-file from what I understand.